pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/ISOMetadataExtractorConfigurationFactory.java

```diff
@@ -143,52 +145,227 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:dataQualityInfo/*/gmd:lineage/*/gmd:statement/gco:CharacterString/text())"));
+
+
+		/*
+			Dates
+		*/
 
 		// Datestamp
 		extractors.add(createXPathExtractor(
 				ResultMetadataFields.DATESTAMP,
 				FieldExtractorType.FIRST_MATCHING_VALUE,
 				"//gmd:MD_Metadata/gmd:dateStamp/*/text()"));
 
+		// Published date
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.DATE_PUBLISHED,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:citation/*/gmd:date/*/gmd:date/gco:Date/text())"));
+
+		// Get all identification info dates
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.DATE_IDENTIFICATION_INFO,
+				new DateXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:citation/*/gmd:date)"));
+
+		// maintenance frequency
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.MAINENANCE_FREQUENCY,
+				new CodeListValueXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:resourceMaintenance/*/gmd:maintenanceAndUpdateFrequency/*[@codeListValue])"));
+
+		/*
+			Organisations
+		 */
 
 		// Organisation names + roles
 		extractors.add(createXPathExtractor(
 				ResultMetadataFields.ORGANISATIONS,
 				new ResponsiblePartyXmlCustomExtractor(),
 				"//gmd:MD_Metadata/gmd:identificationInfo/*/gmd:pointOfContact/gmd:CI_ResponsibleParty"));
 
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_METADATA,
+				new ResponsiblePartyXmlCustomExtractor(),
+				"//gmd:MD_Metadata/gmd:contact/gmd:CI_ResponsibleParty"));
+
+		/*
+			Language data
+		 */
+		// Metadata language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_METADATA,
+				new LanguageXmlCustomExtractor(),
+				"(/csw:GetRecordByIdResponse/*/gmd:language)"));
+
+		//Resource language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_RESOURCE,
+				new LanguageXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:language)"));
+
+
+
+		/*
+		 Constraints
+		*/
+		// use limitation
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_USE_LIMITATION,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:useLimitation/gco:CharacterString/text())"));
+
+		// access limitation
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_ACCESS,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:accessConstraints/gmd:MD_RestrictionCode/@codeListValue)"));
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_OTHER,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:otherConstraints/gco:CharacterString/text())"));
+
+		/*
+			Spatial data
+		 */
+		final String XPATH_CRS_ROOT = "//gmd:referenceSystemInfo/*/gmd:referenceSystemIdentifier";
```

Comment:
Move as public final static to the class

Reply:
Fixed